cli/src/claim_invitation.rs

```diff
@@ -0,0 +1,257 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use clap::Args;
+use std::{path::PathBuf, sync::Arc};
+use terminal_spinners::{SpinnerBuilder, DOTS};
+
+use libparsec::{
+    get_default_config_dir, internal_claimer_retrieve_info as claimer_retrieve_info,
+    BackendInvitationAddr, ClientConfig, DeviceAccessStrategy, HumanHandle, Password,
+    UserOrDeviceClaimInitialCtx,
+};
+
+use crate::utils::*;
+
+#[derive(Args)]
+pub struct ClaimInvitation {
+    /// Parsec config directory
+    #[arg(short, long, default_value_os_t = get_default_config_dir())]
+    config_dir: PathBuf,
+    /// Device slughash
+    #[arg(short, long)]
+    device: Option<String>,
+    /// Server invitation address (e.g.: parsec://127.0.0.1:41905/Org?no_ssl=true&action=claim_user&token=4e45cc21e7604af196173ff6c9184a1f)
+    #[arg(short, long)]
+    addr: BackendInvitationAddr,
+}
+
+pub async fn claim_invitation(claim_invitation: ClaimInvitation) -> anyhow::Result<()> {
```

Comment:
This function is rather long, is it possible to split it by steps ?

Reply:
Done